Currency converters

florent-uzio opened the pull request:
This PR adds two functions to:
1. Convert string to hex with the possibility to convert the string to the non standard currency code for the XRPL. This will be useful for any issued token with the length longer than 3 character.
2. Convert the hex back to a human readable string

Added tests for these functions.

GuillemGarciaDev commented on xrpl/helpers/converters_test.go:
```diff
@@ -0,0 +1,37 @@
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
```
Is this package installed? If not, we can omit it and do comparisons simpler

florent-uzio replied:
Yes it is installed and used in other tests.
What is the easier way to do the comparison ?

GuillemGarciaDev commented on xrpl/helpers/converters.go:
```diff
@@ -0,0 +1,62 @@
+// If isNonStandardCurrency is true, the hexadecimal representation is padded
+// with trailing zeros up to a length of 40 characters. This is to support the non-standard currency codes for the XRPL.
+// See https://xrpl.org/docs/references/protocol/data-types/currency-formats#nonstandard-currency-codes
+func StringToHex(input string, isNonStandardCurrency bool) string {
```
Function should be renamed to `CurrencyStringToHex`

florent-uzio replied:
The thing is that it might be used for something else than currencies, thus the name `StringToHex`.
In xrpl.js we have similar function (starting with `convert`):
<img width="260" alt="image" src="https://github.com/user-attachments/assets/5d1451fd-3d93-4841-874c-3d92f4575eeb">

I can rename it as you suggested and make the functions more dedicated to the currencies only but I wanted to have your opinion first.

AdriaCarrera replied:
@florent-uzio If the function aims to convert a currency code from hex to string then it should be renamed as @GuillemGarciaDev says. However, if it is only a converter from hex to string which has nothing to do with currencies, like the xrpl.js is (https://github.dev/XRPLF/xrpl.js/), then I would suggest leaving the name as it is right now.

As a side note, I would avoid any usage of hex decoded strings in the code and use go native types instead (byte and string)

GuillemGarciaDev commented on xrpl/helpers/converters.go:
```diff
@@ -0,0 +1,62 @@
+}
+
+// HexToString converts a hexadecimal representation to a string.
+func HexToString(input string) (string, error) {
```
Idem

florent-uzio commented:
@GuillemGarciaDev @AdriaCarrera , I updated the functions and focused them on the currency conversion.
So they are now called `CurrencyStringToHex` and `CurrencyHexToString`.

I also moved the two files to the `utils` directory which was created in another PR.
